R/plot-build.R

```diff
@@ -89,6 +89,7 @@ ggplot_build.ggplot <- function(plot) {
   if (npscales$n() > 0) {
     lapply(data, npscales$train_df)
     data <- lapply(data, npscales$map_df)
+    plot$key_data <- npscales$key_data(data)
```

Comment:
Checking the data has to happen before the defaults are filled in and `after_scale()` modifications are applied. Otherwise the data might be lost. I gave the scales list a method to check the data.